lib/app/pages/user_profile_page/user_profile_page.dart

```diff
@@ -95,102 +93,61 @@ class _BodyState extends State<_Body> {
-                    fit: FlexFit.tight,
-                    child: TextField(
-                      decoration:
-                          InputDecoration(labelText: context.tr.userProfilePage_textField, alignLabelWithHint: true),
-                      textAlign: TextAlign.center,
-                      controller: _textController,
-                      maxLength: 10,
-                    ),
-                  ),
-                  Column(
-                    children: [
-                      Padding(
-                        padding: const EdgeInsets.all(AppDimensions.d3),
-                        child: GestureDetector(
-                          onTap: () {
-                            context.read<UserProfileCubit>().onUpdateNameButton(username: _textController.text);
-                            setState(() {
-                              isEditMode = !isEditMode;
-                              _textController.clear();
-                            });
-                          },
-                          child: const Icon(Icons.check),
-                        ),
-                      ),
-                      Padding(
-                        padding: const EdgeInsets.all(AppDimensions.d3),
-                        child: GestureDetector(
-                          onTap: () {
-                            context.read<UserProfileCubit>().onUpdateNameButton(username: _textController.text);
-                            setState(() {
-                              isEditMode = !isEditMode;
-                              _textController.clear();
-                            });
-                          },
-                          child: const Icon(Icons.cancel),
-                        ),
-                      ),
-                    ],
-                  ),
-                ],
+              ChangeUserNameTile(
+                textField: CustomTextField(
+                  controller: _usernameTextController,
+                  maxLength: 10,
+                  hintText: context.tr.userProfilePage_textField,
+                ),
+                buttonToApprove: GestureDetector(
+                  onTap: () {
+                    context.read<UserProfileCubit>().onUpdateUserProfileButton(username: _usernameTextController.text);
+                    setState(
+                      () {
+                        isEditMode = !isEditMode;
+                        _usernameTextController.clear();
+                      },
+                    );
+                  },
+                  child: const Icon(Icons.check),
+                ),
+                buttonToPop: GestureDetector(
+                  onTap: () {
+                    context.router.pop();
+                    setState(() {
+                      isEditMode = !isEditMode;
+                      _usernameTextController.clear();
+                    });
+                  },
+                  child: const Icon(Icons.cancel),
+                ),
```

Comment:
What is the sens of creating a component if you need to provide it's every widget everytime you using it?